File: sdk/ocaml/lib/virtualizor.ml
(** Virtualizor: VPS management via Virtualizor panel *)

(** POST /api/service/{id}/vms/{vmid}/suspend - Suspend virtual server *)
let suspend client params =
  HttpClient.post client "/api/service/{id}/vms/{vmid}/suspend" params

(** POST /api/service/{id}/vms/{vmid}/unsuspend - Unsuspend virtual server *)
let unsuspend client params =
  HttpClient.post client "/api/service/{id}/vms/{vmid}/unsuspend" params

(** GET /api/service/{id}/vms/{vmid}/rebuild - List rebuild templates *)
let list_rebuild_templates client params =
  HttpClient.get client "/api/service/{id}/vms/{vmid}/rebuild" params

(** POST /api/service/{id}/vms/{vmid}/addsshkey - Change SSH key *)
let change_ssh_key client params =
  HttpClient.post client "/api/service/{id}/vms/{vmid}/addsshkey" params

File: sdk/julia/src/virtualizor.jl
# Virtualizor: VPS management via Virtualizor panel
struct VirtualizorResource
    http::HttpClient
end

# POST /api/service/{id}/vms/{vmid}/suspend - Suspend virtual server
function suspend(r::VirtualizorResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/service/{id}/vms/{vmid}/suspend"; params = params)
end

# POST /api/service/{id}/vms/{vmid}/unsuspend - Unsuspend virtual server
function unsuspend(r::VirtualizorResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/service/{id}/vms/{vmid}/unsuspend"; params = params)
end

# GET /api/service/{id}/vms/{vmid}/rebuild - List rebuild templates
function list_rebuild_templates(r::VirtualizorResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/service/{id}/vms/{vmid}/rebuild"; params = params)
end

# POST /api/service/{id}/vms/{vmid}/addsshkey - Change SSH key
function change_ssh_key(r::VirtualizorResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/service/{id}/vms/{vmid}/addsshkey"; params = params)
end
